test/agenda.js

```diff
@@ -3,9 +3,9 @@
 const expect = require('expect.js');
 const {MongoClient} = require('mongodb');
 const delay = require('delay');
-const Job = require('../lib/job');
-const hasMongoProtocol = require('../lib/agenda/has-mongo-protocol');
-const Agenda = require('..');
+const { Job } = require('../dist/job');
```

Comment:
tests should include src files if possbile.
to enable mocka to run typescript files, just add -r ts-node/register  as command line parameter:
```
"mocha": "mocha -r ts-node/register --reporter spec --timeout 8000 -b",
    "mocha-debug": "DEBUG=agenda:**,-agenda:internal:** mocha -r ts-node/register --reporter spec --timeout 8000 -b",
    "mocha-debug-internal": "DEBUG=agenda:internal:** mocha -r ts-node/register --reporter spec --timeout 8000 -b",
    "mocha-debug-all": "DEBUG=agenda:** mocha -r ts-node/register --reporter spec --timeout 8000 -b",
```
and also check Makefile, see
https://github.com/hokify/agenda/blob/typescript/Makefile
https://github.com/hokify/agenda/blob/typescript/package.json
for reference

Reply:
No. Ts node isn’t the same. We should be testing what users get.

Reply:
Wouldn't fully agree with this.
There are two approaches, test early, test late.
Right now (in the old code), as the source file is the same as the packaged file, it is a test early approach. Which means: testing the source files. There are other tests that guarantee that source files are transpiled correctly. Or other tests to see if package building etc works correctly. Therefore sticking to test early would make sense to me. But as always, there are definitely more opinions about this ;) So just my two cents :+1:

Reply:
This is nothing todo with that. Typescript, ts node and compiled JavaScript are all different in their own ways. Adding typescript issues into the mix isn’t going I help devs.

If we need source maps they can be enabled for testing.